[Yukihide Takahashi] iP

yuki-zmstr opened the pull request.

yuki-zmstr committed:
Rename, greet, and exit.

Renamed bot to Yuki, added some greetings.

yuki-zmstr committed:
Add horizontal lines

yuki-zmstr committed:
Add echo function

Says bye when user says bye, echoes otherwise

yuki-zmstr committed:
Add list function

yuki-zmstr committed:
Implement mark as done function

Introduced Task class, add mark done and mark undone functions.

yuki-zmstr committed:
Add ToDos, Events, Deadlines, UI tests

Implement Todo, Event, and Deadlines classes using Inheritance.

Add automated text UI testing

kirangeofran commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+    }
+
+    public static void main(String[] args) {
```
The main method here is quite long and handles multiple responsibilities. Maybe it could be refactored into smaller methods to improve readability.

kirangeofran commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,8 @@
+
+    public Event(String description) {
+        super(description);
+        this.taskType = "[E]";
```
taskType is a bit vague without the context of the question. Perhaps a comment could be added her to explain a bit

kirangeofran commented on src/main/java/Utils.java:
```diff
@@ -0,0 +1,28 @@
+        System.out.println("---------------------------------------------");
+    }
+
+    public static void printWelcomeMessage() {
```
good job on using descriptive method names here !

kirangeofran commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,13 @@
+public class Parser {
```
The use of Parser suggests it's a class for parsing inputs, but its role could maybe explained more clearly through comments or more descriptive method names.

dextboy commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+        }
+        Utils.printExitMessage();
+    }
+}
```
Can consider avoiding long methods as it contain more information than what reader can process at a time. Do consider taking corrective action when it goes beyond 30 LoC.

dextboy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
Good naming of methods in verbs and written in camelCase!

dextboy commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+            command = line.split(" ")[0];
+
+            switch(command) {
+            case "list":
+                System.out.println("Wake up your idea and do these tasks:");
+                for (int i = 0; i < taskCount; i++) {
+                    System.out.println((i + 1) + ".[" + tasks[i].getStatusIcon() + "] "
+                            + tasks[i].taskType + " " + tasks[i].description);
+                }
+                break;
```
Can consider avoiding arrowhead style code.

dextboy commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+public class Yuki {
+
+    // create array of Tasks (use of polymorphism)
+    private static Task[] tasks = new Task[100];
```
Can avoid magic numbers and use named constants instead.

dextboy replied:
Good that array specifiers are attached to the type and not the variable!

dextboy commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+    public static void main(String[] args) {
+        Utils.printWelcomeMessage();
+
+        // accept user input
```
Can consider not repeating the obvious for comments.

dextboy commented on src/main/java/Yuki.java:
```diff
@@ -0,0 +1,87 @@
+                    System.out.println((i + 1) + ".[" + tasks[i].getStatusIcon() + "] "
+                            + tasks[i].taskType + " " + tasks[i].description);
+                }
+                break;
+
+            case "mark":
+                index_task = Integer.parseInt(line.split(" ")[1]) - 1;
+                tasks[index_task].markAsDone();
+                break;
+
+            case "unmark":
+                index_task = Integer.parseInt(line.split(" ")[1]) - 1;
+                tasks[index_task].markAsUndone();
+                break;
+
+            case "todo":
+                data = Parser.parseInput(line.substring(command.length()));
+                description = data[0];
+                addTask(new Todo(description));
+                reportNumberOfTasks();
+                break;
+
+            case "deadline":
+                data = Parser.parseInput(line.substring(command.length()));
+                description = data[0] + " (by:" + data[1] + ")";
+                addTask(new Deadline(description));
+                reportNumberOfTasks();
+                break;
+
+            case "event":
+                data = Parser.parseInput(line.substring(command.length()));
+                description = data[0] + " (from: " + data[1] + " to: " + data[2] + ")";
+                addTask(new Event(description));
+                reportNumberOfTasks();
+                break;
+
+            default:
+                Utils.printInstructions();
+                break;
+            }
```
Good that there is no indentation for case clauses.

yuki-zmstr committed:
Review comments on PR

Fix code quality and readability issues.

yuki-zmstr committed:
Handle invalid command and description errors

Print warning message and instructions when command is not recognized or description is not valid for that command.

yuki-zmstr committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Handle invalid command and description errors
  Review comments on PR

yuki-zmstr committed:
Organise into packages

yuki-zmstr committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Organise into packages

yuki-zmstr committed:
Add print statements to match UI test output

Accidentally deleted print statements in add{Task} methods, re-added.

yuki-zmstr committed:
Merge branch 'fix-text-ui-test'

* fix-text-ui-test:
  Add print statements to match UI test output

yuki-zmstr committed:
Organise package structure

Shift Yuki.java into yuki package and update runtest.sh

yuki-zmstr committed:
Use ArrayList to store tasks

yuki-zmstr committed:
Refactor mark and unmark function

Override Exceptions toString also.

yuki-zmstr committed:
Refactor constants

yuki-zmstr committed:
Add delete functionality

yuki-zmstr committed:
Add read and write functionality

yuki-zmstr committed:
Add default tasks

yuki-zmstr committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Add delete functionality
  Refactor constants
  Refactor mark and unmark function
  Use ArrayList to store tasks

yuki-zmstr committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add default tasks
  Add read and write functionality

# Conflicts:
#	src/main/java/yuki/Yuki.java

yuki-zmstr committed:
Add help entry in instructions

yuki-zmstr committed:
Use e.getMessage() to display error message

yuki-zmstr committed:
Update default tasks

yuki-zmstr committed:
Add manifest version 1.0

yuki-zmstr committed:
fix delete bug

Added 'break' in case statement

wenxin-c commented on src/main/java/yuki/Yuki.java:
```diff
@@ -0,0 +1,229 @@
+    private static final String PATH_TO_FILE = "data/tasks.txt";
+
+    public static void readFile() throws FileNotFoundException {
+        File f = new File(PATH_TO_FILE);
+        Scanner s = new Scanner(f);
```
Try to give variables a more meaningful name instead of a single alphabet.

wenxin-c commented on src/main/java/yuki/Yuki.java:
```diff
@@ -0,0 +1,229 @@
+    public static void writeFile() throws IOException {
+        File dataDirectory = new File("data");
+        if (!dataDirectory.exists()) {
+            boolean directoryCreated = dataDirectory.mkdirs(); // create directory to save in.
```
Try to name boolean variables sound more boolean, e.g. hasDirectoryCreated.

wenxin-c commented on src/main/java/yuki/Yuki.java:
```diff
@@ -0,0 +1,229 @@
+
+        String command;
+
+        while (!line.equals(Constants.EXIT_COMMAND)) {
```
This while loop is very long, consider abstracting it further to break it down.

yuki-zmstr replied:
In the TaskList.java > executeCommand method, I would like to reduce the number of try, catch statents - do you have advice on how I can do so?

![Screenshot 2024-02-24 at 8 50 35 PM](https://github.com/nus-cs2113-AY2324S2/ip/assets/72074553/186213aa-3398-439f-a17e-0887c886cbb1)

wenxin-c commented on src/main/java/yuki/Yuki.java:
```diff
@@ -0,0 +1,229 @@
+        String line;
+        while (s.hasNext()) {
+            line = s.nextLine();
+            if (line.charAt(7) == 'T') {
```
Try not to use magic literals e.g. magic numbers and magic string, consider creating a constant and use the constant in your code instead.

wenxin-c reviewed (commented):
Well done in your ip so far. Your code is well done with only some minor detail need to be more careful of.

yuki-zmstr committed:
Rename variables and extract magic literals

yuki-zmstr committed:
Add Storage class to handle read and write

yuki-zmstr committed:
Add feedback to user when written successfully

yuki-zmstr committed:
Refactor Input Parser

yuki-zmstr committed:
Set constants to public

yuki-zmstr committed:
Abstract out TaskList and UI

yuki-zmstr committed:
Make storage a local variable in Constructor

yuki-zmstr committed:
Extract magic number

yuki-zmstr committed:
Fix bug when help command entered

yuki-zmstr committed:
greetUser and sayByeToUser methods in UI

yuki-zmstr committed:
Add boolean parameter isDone to Task

yuki-zmstr committed:
Fix bug not reading completion status from input file

yuki-zmstr committed:
Add method to get description

yuki-zmstr committed:
Extract Add task message to user

yuki-zmstr committed:
Function to find task by keyword

yuki-zmstr committed:
Add JavaDoc to important public classes and methods

yuki-zmstr committed:
Catch ArrayIndexOutOfBoundsException

Catch error when user does not input an index in mark, unmark, delete command.

yuki-zmstr committed:
Merge pull request #1 from yuki-zmstr/branch-A-JavaDoc

Add JavaDoc to important public classes and methods

yuki-zmstr committed:
Merge branch 'master' of https://github.com/yuki-zmstr/ip

* 'master' of https://github.com/yuki-zmstr/ip:
  Catch ArrayIndexOutOfBoundsException
  Add JavaDoc to important public classes and methods

yuki-zmstr committed:
Resolve merge conflict

yuki-zmstr committed:
Merge pull request #2 from yuki-zmstr/branch-Level-9

Add function to search for a task by keyword

yuki-zmstr committed:
Merge branch 'master' of https://github.com/yuki-zmstr/ip

* 'master' of https://github.com/yuki-zmstr/ip:
  Function to find task by keyword
  Extract Add task message to user
  Add method to get description

yuki-zmstr committed:
Add User Guide

yuki-zmstr committed:
Update text UI testing

yuki-zmstr committed:
Update docs/README.md

yuki-zmstr committed:
Add JavaDoc to methods from branch-Level-9, refactor

yuki-zmstr committed:
Update User Guide

yuki-zmstr committed:
Update User Guide to have example of each command

yuki-zmstr committed:
Add simple CLI instructions for delete and find

yuki-zmstr committed:
In user guide, add command to execute JAR file

yuki-zmstr committed:
Add more javaDoc and edit comments
